modules/sequenza/1.0/sequenza.smk

```diff
@@ -0,0 +1,226 @@
+    conda:
+        CFG["conda_envs"]["sequenza"]
+    shell:
+        "sequenza-utils bam2seqz --qlimit 30 -gc {input.sequenza_gc} --fasta {input.genome} "
+        "-n {input.normal_bam} -t {input.tumour_bam} --chromosome {wildcards.chr} | sequenza-utils "
+        "seqz_binning -w 300 -s - | gzip > {output}"
+
+merged_string = CFG["dirs"]["sequenza"] + "{{seq_type}}--{{genome_build}}/{{tumour_id}}--{{normal_id}}--{{pair_status}}/{{tumour_id}}_{chr}.binned.out.seqz.gz"
+
+rule _sequenza_merge_seqz:
+    input:
+        expand(merged_string,chr=CFG["chroms"])
+    output:
+        merged_seqz = CFG["dirs"]["sequenza"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.merged.seqz.gz"
+    params:
+        merge = CFG["options"]["merge_seqz"]
+    log:
+        stdout = CFG["logs"]["sequenza"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_bam2seqz.stdout.log",
+        stderr = CFG["logs"]["sequenza"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_bam2seqz.stderr.log"
+    resources: mem_mb = 10000
+    shell:
+        "bash {params.merge} {input} | gzip > {output.merged_seqz}"
+
+
+rule _sequenza_filter_seqz:
+    input:
+        merged_seqz = rules._sequenza_merge_seqz.output.merged_seqz
+    output:
+        filtered_seqz = CFG["dirs"]["seqz_filtered"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}.filtered.seqz.gz"
+    log:
+        stdout = CFG["logs"]["sequenza"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_filter_seqz.stdout.log",
+        stderr = CFG["logs"]["sequenza"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_filter_seqz.stderr.log"
+    resources: mem_mb = 20000
+    threads: 1
+    params:
+        dbsnp_pos = "reference/genomes/{genome_build}/annotations/{genome_build}.dbsnp.pos.sort", #fix this to use the path to the reference files based on the config
+        filter_seqz = CFG["options"]["filter_seqz"]
+    shell:
+        "{params.filter_seqz} {input.merged_seqz} {params.dbsnp_pos} | gzip > {output.filtered_seqz}"
+
+rule _sequenza_unfiltered_analysis:
+    input:
+        merged_seqz = rules._sequenza_merge_seqz.output.merged_seqz,
+        normal_bam = CFG["dirs"]["inputs"] + "bam/{seq_type}--{genome_build}/{normal_id}.bam"
+    output:
+        segments = CFG["dirs"]["plots"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_unfiltered_segments.txt"
+    log:
+        stdout = CFG["logs"]["plots"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_unfiltered_analysis.stdout.log",
+        stderr = CFG["logs"]["plots"] + "{seq_type}--{genome_build}/{tumour_id}--{normal_id}--{pair_status}/{tumour_id}_sequenza_unfiltered_analysis.stderr.log"
+    resources: mem_mb = 100000
```

Comment:
Does this step really need 100 GB of memory?

Reply:
I notice that `_sequenza_filtered_analysis` has the same thing.

Reply:
Doubtful I'll test it with a genome and exome to see what is reasonable.

Reply:
Peak memory usage was less than 20G for the single genome I tested. As long as we filter the SNVs, it shouldn't be worse for FFPE. The unfiltered analysis could be much worse for FFPE. I think we should reduce this to 20G and see how it goes. We can up it or do something like the following if necessary:

mem_mb=lambda wildcards, attempt: attempt * 20000